test/tools/cmap_spec_runner.ts

```diff
@@ -241,16 +247,22 @@ export class ThreadContext {
   #poolOptions: Partial<ConnectionPoolOptions>;
   #hostAddress: HostAddress;
   #supportedOperations: ReturnType<typeof getTestOpDefinitions>;
+  #injectPoolStats = false;
 
   /**
    *
    * @param hostAddress - The address of the server to connect to
    * @param poolOptions - Allows the test to pass in extra options to the pool not specified by the spec test definition, such as the environment-dependent "loadBalanced"
    */
-  constructor(hostAddress: HostAddress, poolOptions: Partial<ConnectionPoolOptions> = {}) {
+  constructor(
+    hostAddress: HostAddress,
+    poolOptions: Partial<ConnectionPoolOptions> = {},
+    contextOptions: { injectPoolStats: boolean }
+  ) {
     this.#poolOptions = poolOptions;
     this.#hostAddress = hostAddress;
     this.#supportedOperations = getTestOpDefinitions(this);
+    this.#injectPoolStats = contextOptions.injectPoolStats;
```

Comment:
```suggestion
this.#injectPoolStats ??= contextOptions.injectPoolStats;
```

The class default will always be overriden, and the spec test leaves this option undefined. Nothing breaks becasue of faleyness but might as well make sure this is set correctly.

Reply:
it's actually because of the runner - look in line 456, but no harm with the extra guard

Reply:
Actually, this particular guard would be not what we want, since it would prevent overriding the default... the nullish coalescing compares the left hand side, not the right hand side; if you'd like, I can set it to `!!contextOptions.injectPoolStats` here and remove the defaulting on the class